examples/oneview_server_profile_facts.yml

```diff
@@ -0,0 +1,95 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+---
+- hosts: all
+  vars:
+    config: "{{ playbook_dir }}/oneview_config.json"
+    # set the name of an existent server profile to run this example
+    server_profile_name: "Demo Server Profile"
+    # set the uri of an existent enclosure group to run this example
+    enclosure_group_uri: '/rest/enclosure-groups/3af25c76-dec7-4753-83f6-e1ad06c29a43'
+    # set the uri of an existent server hardware type to run this example
+    server_hardware_type_uri: '/rest/server-hardware-types/C8DEF9A6-9586-465E-A951-3070988BC226'
+    # set the ID of an existent storage system to run this example
+    storage_system_id: "TXQ1010307"
+  tasks:
+    - name: Gather facts about all Server Profiles
+      oneview_server_profile_facts:
+        config: "{{ config }}"
+      delegate_to: localhost
+
+    - debug: msg="{{server_profiles | map(attribute='name') | list }}"
+
+    - name: Gather facts about a Server Profile by name
+      oneview_server_profile_facts:
+        config: "{{ config }}"
+        name: "{{server_profile_name}}"
+      delegate_to: localhost
+
+    - debug: var=server_profiles
+
+
+    - name: Gather facts about available servers and bays for a given enclosure group and server hardware type
+      oneview_server_profile_facts:
+        config: "{{ config }}"
+        options:
+          - availableTargets:
+              serverHardwareTypeUri: "{{server_hardware_type_uri}}"
+              enclosureGroupUri: "{{enclosure_group_uri}}"
+      delegate_to: localhost
+
+    - debug: var=server_profile_available_targets
+
+    - name: Gather all facts about a Server Profile
+      oneview_server_profile_facts:
+       config: "{{ config }}"
+       name : "{{server_profile_name}}"
+       options:
+           - schema
+           - compliancePreview
+           - profilePorts:
+              serverHardwareTypeUri: "{{server_hardware_type_uri}}"
+              enclosureGroupUri: "{{enclosure_group_uri}}"
+           - messages
+           - transformation:
+               serverHardwareTypeUri: "{{server_hardware_type_uri}}"
+               enclosureGroupUri: "{{enclosure_group_uri}}"
+           - availableNetworks:
+               serverHardwareTypeUri: "{{server_hardware_type_uri}}"
+               enclosureGroupUri: "{{enclosure_group_uri}}"
+           - availableServers
+           - availableStorageSystem:
+               storageSystemId: "{{storage_system_id}}"
+               serverHardwareTypeUri: "{{server_hardware_type_uri}}"
+               enclosureGroupUri: "{{enclosure_group_uri}}"
+           - availableStorageSystems:
+               serverHardwareTypeUri: "{{server_hardware_type_uri}}"
+               enclosureGroupUri: "{{enclosure_group_uri}}"
+           - availableTargets
+
+      delegate_to: localhost
+
+    - debug: var=server_profiles
```

Comment:
It is necessary to output the facts returned by the module. As this resource has an array of methods to retrieve information about server profiles, and this example includes all the possible options, then it is necessary all this debug to output all returned data